ios/browser/api/bookmarks/brave_bookmarks_api.mm

```diff
@@ -55,11 +58,66 @@ - (instancetype)initWithNode:(const bookmarks::BookmarkNode*)node
   if ((self = [super init])) {
     self->node_ = node;
     self->model_ = model;
+    self->owned_ = false;
+  }
+  return self;
+}
+
+- (instancetype)initWithTitle:(NSString*)title
+                           id:(int64_t)id
+                         guid:(NSString*)guid
+                          url:(NSURL*)url
+                    dateAdded:(NSDate*)dateAdded
+                 dateModified:(NSDate*)dateModified
+                     children:
+                         (NSArray<IOSBookmarkNode*>*)children {
+  if ((self = [super init])) {
+    //Only in a NEWER version of Chromium, they have `base::GUID`
+    std::string guid_; //base::GUID guid_ = base::GUID();
+    int64_t id_ = static_cast<int64_t>(id);
+    
+    if ([guid length] > 0) {
+      DCHECK(base::IsValidGUID(base::SysNSStringToUTF16(guid)));
+      guid_ = [guid UTF8String];
+    } else {
+      guid_ = base::GenerateGUID();
+    }
+    
+    GURL gurl_ = net::GURLWithNSURL(url);
+    bookmarks::BookmarkNode* node = new bookmarks::BookmarkNode(id_, guid_, gurl_);
+    node->SetTitle(base::SysNSStringToUTF16(title));
+    
+    if (dateAdded) {
+      node->set_date_added(base::Time::FromDoubleT([dateAdded timeIntervalSince1970]));
+    }
+    
+    if (dateModified) {
+      node->set_date_folder_modified(base::Time::FromDoubleT([dateModified timeIntervalSince1970]));
+    }
+    
+    for (IOSBookmarkNode* child : children) {
+      DCHECK(child->owned_); //Bookmark must be owned and non-const. IE: Allocated from iOS/Swift.
+      child->owned_ = false;
+      child->node_ =
+              node->Add(std::unique_ptr<bookmarks::BookmarkNode>(const_cast<bookmarks::BookmarkNode*>(child->node_)));
+    }
+    
+    self->node_ = node;
+    self->model_ = nil;
+    self->owned_ = true;
   }
   return self;
 }
 
 - (void)dealloc {
+  if (self->owned_) {
+    // All Objective-C++ class pointers are reference counted.
+    // No need for copy or move constructor & assignment operators.
+    // They're automatic RAII.. So we can safely delete an owned raw pointer here.
+    delete self->node_;
```

Comment:
this is definitely not right. https://github.com/brave/brave-core/pull/6722/commits/41766d1264a67b8756068d958a5c8240403d9527#diff-41abb926a5a9b81d6d2cb900a29b1f2aba295cea9efd99ba9e788a6d0030d133R48 should be marked as NOT OWNED and this definitely should not be deleted